functions/New-DbaSqlParameter.ps1

```diff
@@ -0,0 +1,215 @@
+
+    .PARAMETER TypeName
+        Sets the type name for a table-valued parameter.
+
+    .PARAMETER UdtTypeName
+        Sets a string that represents a user-defined type as a parameter.
+
+    .PARAMETER Value
+        Sets the value of the parameter.
+
+    .PARAMETER EnableException
+        By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+        This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+        Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .NOTES
+        Tags: Query
+        Author: Chrissy LeMaire (@cl), netnerds.net
+        Website: https://dbatools.io
+        Copyright: (c) 2021 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/New-DbaSqlParameter
+
+    .EXAMPLE
+        PS C:\> New-DbaSqlParameter -ParameterName json_result -SqlDbType NVarChar -Size -1 -Direction Output
+
+        Creates a SqlParameter object that can be used with Invoke-DbaQuery
+
+    .EXAMPLE
+        PS C:\> $output = New-DbaSqlParameter -ParameterName json_result -SqlDbType NVarChar -Size -1 -Direction Output
+        PS C:\> Invoke-DbaQuery -SqlInstance localhost -Database master -CommandType StoredProcedure -Query my_proc -SqlParameters $output
+        PS C:\> $output.Value
+
+        Creates an output parameter and uses it to invoke a stored procedure.
+    #>
+    [CmdletBinding()]
+    param(
+        [ValidateSet("None", "IgnoreCase", "IgnoreNonSpace", "IgnoreKanaType", "IgnoreWidth", "BinarySort2", "BinarySort")]
+        [string]$CompareInfo,
+        [ValidateSet("AnsiString", "Binary", "Byte", "Boolean", "Currency", "Date", "DateTime", "Decimal", "Double", "Guid", "Int16", "Int32", "Int64", "Object,
+        SByte", "Single", "String", "Time", "UInt16", "UInt32", "UInt64", "VarNumeric", "AnsiStringFixedLength", "StringFixedLength", "Xml,
+        DateTime2", "DateTimeOffset")]
+        [string]$DbType,
+        [ValidateSet("Input", "Output", "InputOutput", "ReturnValue")]
+        [string]$Direction,
+        [switch]$ForceColumnEncryption,
+        [switch]$IsNullable,
+        [int]$LocaleId,
+        [string]$Offset,
+        [Alias("Name")]
+        [string]$ParameterName,
+        [string]$Precision,
+        [string]$Scale,
+        [int]$Size,
+        [string]$SourceColumn,
+        [switch]$SourceColumnNullMapping,
+        [ValidateSet("Original", "Current", "Proposed", "Default")]
+        [string]$SourceVersion,
+        [ValidateSet("BigInt", "Binary", "Bit", "Char", "DateTime", "Decimal", "Float", "Image", "Int", "Money", "NChar", "NText", "NVarChar", "Real,
+        UniqueIdentifier", "SmallDateTime", "SmallInt", "SmallMoney", "Text", "Timestamp", "TinyInt", "VarBinary", "VarChar", "Variant", "Xml,
```

Comment:
not sure this is indented correctly

Reply:
you were right! thanks, that was super messed up.